nvflare/app_common/workflows/cyclic_ctl.py

```diff
@@ -264,11 +265,12 @@ def restore(self, state_data: dict, fl_ctx: FLContext):
         finally:
             pass
 
-    def handle_dead_job(self, client_name: str, fl_ctx: FLContext):
-        super().handle_dead_job(client_name, fl_ctx)
-
-        new_client_list = []
-        for client in self._participating_clients:
-            if client_name != client.name:
-                new_client_list.append(client)
-        self._participating_clients = new_client_list
+    def handle_event(self, event_type, fl_ctx):
+        super().handle_event(event_type, fl_ctx)
```

Comment:
This is good, just a note, if we have this, for the same event, if parent has some logic to handle as well, we will be handling the same event both in parent and in child.

Reply:
Right we could also call this in an `else` if we want

Reply:
right, it depends on the situations, right now it is good as it is .

Reply:
Once we remove the handle_event from controller.py, then we won't have this issue.

Reply:
removed the handle_event from controller.py, added additional `InfoCollector.EVENT_TYPE_SET_STATS` since `collector.set_info()` must occur before `collector.add_info()`